### PR TITLE
feat: [PIE-5177]: add support for multi services

### PR DESCRIPTION
**Pull request by vkbansal-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

## [Contributor license agreement](https://github.com/harness/harness-core-ui/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

### CODE REVIEW

File: src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/DeployServiceEntityWidget.tsx [hunk 1]
```diff
@@ -0,0 +1,551 @@
...
+    close: closeSwitchToSingleSvcDialog
+  } = useToggleOpen()
+  const [allServices, setAllServices] = useState(getAllFixedServices(initialValues))
+  const { MULTI_SERVICE_INFRA } = useFeatureFlags()
+
+  const {
+    data: servicesListResponse,
+    error,
+    loading: loadingServicesList
+  } = useGetServiceList({
```
Comment (AshwinBhatkal-Harness) [inline src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/DeployServiceEntityWidget.tsx:156]: We are making the get service list call in the invocation map of deployServiceEntity step and here as well. We can avoid one call I guess

Comment (vkbansal-harness) [reply src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/DeployServiceEntityWidget.tsx:156]: Ok will try to fix this in follow up PR

**Comment by vkbansal-harness** [comment]
retrigger cypress

**Comment by vkbansal-harness** [comment]
trigger cypress

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**